# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: gfx/wr/webrender/src/render_task.rs
# Commit: 7b5ad213cf68
# Full Hash: 7b5ad213cf68b889c96bf5261bf21ca9a20fccf3
# Author: Ashley Hale <ahale@mozilla.com>
# Date: 2025-04-11 03:05:25
# Description:
#   Bug 1924241 - Use a minimum task size for SVGFEDropShadow and SVGFEGaussianBlur filters to avoid panic r=gfx-reviewers,nical
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D244698
# ==============================================================================

diff -r efe56d7cba09 -r 7b5ad213cf68 gfx/wr/webrender/src/render_task.rs
--- a/gfx/wr/webrender/src/render_task.rs	Thu Apr 10 14:10:17 2025 +0000
+++ b/gfx/wr/webrender/src/render_task.rs	Thu Apr 10 14:13:12 2025 +0000
@@ -2294,13 +2294,16 @@
                             adjusted_blur_std_deviation.width * BLUR_SAMPLE_SCALE,
                             adjusted_blur_std_deviation.height * BLUR_SAMPLE_SCALE)
                         .round_out();
-                    let blur_task_size = blur_subregion.size().cast_unit();
+                    let blur_task_size = blur_subregion
+                        .size()
+                        .cast_unit()
+                        .max(DeviceSize::new(1.0, 1.0));
                     // Adjust task size to prevent potential sampling errors
                     let adjusted_blur_task_size =
                         BlurTask::adjusted_blur_source_size(
                             blur_task_size,
                             adjusted_blur_std_deviation,
-                        ).to_f32();
+                        ).to_f32().max(DeviceSize::new(1.0, 1.0));
                     // Now change the subregion to match the revised task size,
                     // keeping it centered should keep animated radius smooth.
                     let corner = LayoutPoint::new(
@@ -2412,13 +2415,16 @@
                             adjusted_blur_std_deviation.width * BLUR_SAMPLE_SCALE,
                             adjusted_blur_std_deviation.height * BLUR_SAMPLE_SCALE)
                         .round_out();
-                    let blur_task_size = blur_subregion.size().cast_unit();
+                    let blur_task_size = blur_subregion
+                        .size()
+                        .cast_unit()
+                        .max(DeviceSize::new(1.0, 1.0));
                     // Adjust task size to prevent potential sampling errors
                     let adjusted_blur_task_size =
                         BlurTask::adjusted_blur_source_size(
                             blur_task_size,
                             adjusted_blur_std_deviation,
-                        ).to_f32();
+                        ).to_f32().max(DeviceSize::new(1.0, 1.0));
                     // Now change the subregion to match the revised task size,
                     // keeping it centered should keep animated radius smooth.
                     let corner = LayoutPoint::new(
